CHANGELOG.md

```diff
@@ -82,8 +82,49 @@
   exist with `gleam add`.
   ([Giacomo Cavalieri](https://github.com/giacomocavalieri))
 
-- External files (such as `.mjs` and `.erl`) are now permitted in subdirectories
-  of `src/` and `test/`.
+- The compiler can now suggest to wrap a value in an `Ok` or `Error` if that can
+  solve a type mismatch error:
+
+  ```gleam
+  pub fn first(list: List(a)) -> Result(a, Nil) {
+    case number {
+      [] -> Error(Nil)
+      [first, ..rest] -> first
+    }
+  }
+  ```
+
+  Results in the following error:
+
+  ```txt
+  error: Type mismatch
+    ┌─ /src/one/two.gleam:5:5
+    │
+  5 │     [first, ..rest] -> first
+    │     ^^^^^^^^^^^^^^^^^^^^^^^^
+    │                        │
+    │                        Did you mean to wrap this in an `Ok`?
```

Comment:
This error is wrong! `Ok([first, ..rest] -> first)` would not be valid

Reply:
The message is referring to the `first` bit but the way the two overlapping spans are rendered it's not very clear; do you think we should stop highlighting the entire case branch in this case? However that would mean that the error message is not very precise now:
```
[first, ..rest] -> first
                   ^^^^^ Did you mean to wrap this in an `Ok`?

This case clause was found to return a different type than the previous
one, but all case clauses must return the same type.
``` 
And it would be more confusing when the part on the right of `->` is on its own line as the clause would not be displayed in the error!
```
  first
  ^^^^^ Did you mean to wrap this in an `Ok`?

This case clause was found to return a different type than the previous
one, but all case clauses must return the same type.
```

Reply:
I think we should continue to highlight the entire cause as the source of the type error, but a label about wrapping an expression should only point to the code that is to be wrapped.

Reply:
The tricky thing is it is already pointing to just the code that needs to be wrapped but the way the library renders it it's not immediately obvious. If the then clause were not on the same line what you'd see is this:
```gleam
     case wibble {
       A -> Error("a")
  ->   wobble ->
 |     1
 |     ^ Did you mean to wrap this ...
 |
 incorrect branch
```
But when the two are on the same line the two look like this
```gleam
wobble -> 1
^^^^^^^^^^^
          |
          Did you mean to wrap this
incorrect branch
```
The way it gets rendered is not the best, maybe I could try and remove it for branches when the then clause is on the same line or for case matches entirely

Reply:
Ah, I see, that is annoying.